apm-collector/apm-collector-agentstream/src/main/java/org/skywalking/apm/collector/agentstream/worker/segment/buffer/SegmentBufferReader.java

```diff
@@ -0,0 +1,151 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Project repository: https://github.com/OpenSkywalking/skywalking
+ */
+
+package org.skywalking.apm.collector.agentstream.worker.segment.buffer;
+
+import com.google.protobuf.CodedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.skywalking.apm.collector.agentstream.worker.segment.SegmentParse;
+import org.skywalking.apm.collector.core.util.CollectionUtils;
+import org.skywalking.apm.collector.core.util.Const;
+import org.skywalking.apm.collector.core.util.StringUtils;
+import org.skywalking.apm.network.proto.UpstreamSegment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author pengys5
+ */
+public enum SegmentBufferReader {
+    INSTANCE;
+
+    private final Logger logger = LoggerFactory.getLogger(SegmentBufferReader.class);
+    private InputStream inputStream;
+
+    public void initialize() {
+        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(this::preRead, 3, 3, TimeUnit.SECONDS);
+    }
+
+    private void preRead() {
+        String readFileName = OffsetManager.INSTANCE.getReadFileName();
+        if (StringUtils.isNotEmpty(readFileName)) {
+            File readFile = new File(SegmentBufferConfig.BUFFER_PATH + readFileName);
+            if (readFile.exists()) {
+                deleteTheDataFilesBeforeReadFile(readFileName);
+                long readFileOffset = OffsetManager.INSTANCE.getReadFileOffset();
+                read(readFile, readFileOffset);
+                readEarliestCreateDataFile();
+            } else {
+                deleteTheDataFilesBeforeReadFile(readFileName);
+                readEarliestCreateDataFile();
+            }
+        } else {
+            readEarliestCreateDataFile();
+        }
+    }
+
+    private void deleteTheDataFilesBeforeReadFile(String readFileName) {
+        File[] dataFiles = new File(SegmentBufferConfig.BUFFER_PATH).listFiles(new PrefixFileNameFilter());
+
+        long readFileCreateTime = getFileCreateTime(readFileName);
+        for (File dataFile : dataFiles) {
+            long fileCreateTime = getFileCreateTime(dataFile.getName());
+            if (fileCreateTime < readFileCreateTime) {
+                dataFile.delete();
+            } else if (fileCreateTime == readFileCreateTime) {
+                break;
+            }
+        }
+    }
+
+    private long getFileCreateTime(String fileName) {
```

Comment:
A suggestion only, is using long as a time representation a good idea? In my olde time, I prefer, `YYYYMMDDHHmmSS` to represent time.